app/celer-export-geant.cc

```diff
@@ -76,6 +78,55 @@ GeantPhysicsOptions load_options(std::string const& option_filename)
     return options;
 }
 
+//---------------------------------------------------------------------------//
+void run(std::string const& gdml_filename,
+         std::string const& opts_filename,
+         std::string const& out_filename)
+{
+    // Construct options, set up Geant4, read data
+    auto imported = [&] {
+        GeantImporter import(
+            GeantSetup(gdml_filename, load_options(opts_filename)));
+        GeantImporter::DataSelection selection;
+        selection.particles = GeantImporter::DataSelection::em
+                              | GeantImporter::DataSelection::optical;
+        selection.processes = selection.particles;
+        selection.reader_data = true;
+        return import(selection);
+    }();
+
+    ScopedRootErrorHandler scoped_root_error;
+
+    if (ends_with(out_filename, ".root"))
+    {
+        // Write ROOT file
+        RootExporter export_root(out_filename.c_str());
+        export_root(imported);
+    }
+    else if (ends_with(out_filename, ".json"))
+    {
+        // Write JSON to file
+        CELER_LOG(info) << "Opening JSON output at " << out_filename;
+        std::ofstream os(out_filename);
+        RootJsonDumper dump_root(&os);
```

Comment:
```suggestion
        RootJsonDumper export_json(&os);
```
so it reads more naturally compared to the previous `if` statement block.